### ISSUE TITLE
Benchmark & investigate: Warm restart cosine annealing uses exponential decay instead of cosine schedule

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#1656]
## Context

Part of #1649

In `src/propagate/BackPropagation.ts` line 257, the warm restart strategy uses exponential decay within each period:

```typescript
case "warm_restart":
  const positionInPeriod = iteration % period;
  return config.initialLearningRate * Math.pow(config.learningRateDecay, positionInPeriod);
```

Standard cosine annealing with warm restarts (SGDR / Loshchilov & Hutter 2017) uses a cosine schedule:

```
lr = lr_min + 0.5 * (lr_max - lr_min) * (1 + cos(π * T_cur / T_i))
```

The cosine schedule decays slowly at first (allowing exploration), accelerates in the middle, and slows near the minimum (fine-tuning), before resetting. The current exponential decay drops quickly and uniformly, spending less time at useful learning rates.

For production-sized creatures where convergence is slow, spending more time at moderate learning rates could improve the quality of updates.

## What needs to be done

1. Create a benchmark comparing convergence of exponential warm restart vs cosine annealing warm restart on a production-sized creature
2. If cosine annealing shows better convergence, implement it as the warm restart schedule
3. If no meaningful improvement, close as negative result with benchmark data

## Acceptance criteria

- [ ] Benchmark comparing both schedules on production-sized creature
- [ ] If improved: cosine annealing implementation with passing learning rate scheduling tests
- [ ] If no gain: issue closed with benchmark results

## Dependencies

None — can be worked on independently.

## Approach

This is a benchmark investigation as specified in the parent issue.
